### PR TITLE
Split azure-security-keyvault

### PR DESCRIPTION
**Pull request by chlowell**
`azure-security-keyvault` becomes `azure-keyvault-keys` and `azure-keyvault-secrets`. One important consequence is that the track 1 library, `azure-keyvault`, can't coexist with these new packages.

@samvaity please look over the docs, I may have missed something

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by chlowell** [comment]
@scbedd this can't pass CI if we test everything under `/sdk/keyvault`. Can we narrow the scope such that `azure-keyvault` isn't involved when we touch `azure-keyvault-*`?

**Comment by chlowell** [comment]
Nevermind, we'll remove `azure-keyvault` instead.

File: sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py [hunk 1]
```diff
@@ -19,7 +18,7 @@ def test_create_key_client():
     # [START create_key_client]
 
     from azure.identity.aio import DefaultAzureCredential
-    from azure.security.keyvault.aio import KeyClient
+    from azure.keyvault.keys.aio import KeyClient
 
     # Create a KeyClient using default Azure credentials
     credential = DefaultAzureCredential()
```
Comment (samvaity) [inline sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py]: I think this is existing but needs to be updated to `DefaultAzureCredential()` ?

File: sdk/keyvault/azure-keyvault-keys/samples/list_operations.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-from azure.security.keyvault import KeyClient
+from azure.keyvault.keys import KeyClient
 from azure.identity import DefaultAzureCredential
 from azure.core.exceptions import HttpResponseError
 
- # ----------------------------------------------------------------------------------------------------------
+# ----------------------------------------------------------------------------------------------------------
 # Prerequistes -
 #
 # 1. An Azure Key Vault-
```
Comment (samvaity) [inline sdk/keyvault/azure-keyvault-keys/samples/list_operations.py]: Need to remove `security` from the package name on point `2` and update the name also to reflect the  corresponding package name in all the samples files.
>  2. Microsoft Azure Key Vault PyPI package -
> https://pypi.python.org/pypi/azure-security-keyvault/

File: sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py (hunk 1 above)
Comment (samvaity) [reply sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py]: Nevermind, I guess this was an intentional update.

File: sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py [hunk 3]
```diff
@@ -21,6 +20,7 @@ class KeyClient(_KeyVaultClientBase):
             :end-before: [END create_key_client]
             :language: python
             :caption: Creates a new instance of the Key client
+            :dedent: 4
```
Comment (samvaity) [inline sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py]: we are adding back the dedents and now they are different across the file?

File: sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py (hunk 1 above)
Comment (chlowell) [reply sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py]: I don't follow, this is already using `DefaultAzureCredential`?

File: sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py (hunk 3 above)
Comment (chlowell) [reply sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py]: Yep, I rendered the docs and saw class examples were one level too deep, method examples two levels too deep.

File: sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py (hunk 1 above)
Comment (samvaity) [reply sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py]: Oh, I meant `AsyncDefaultAzureCredential`

File: sdk/keyvault/azure-keyvault-keys/samples/hello_world_async.py [hunk 4]
```diff
@@ -12,12 +12,12 @@
 #    https://docs.microsoft.com/en-us/azure/key-vault/quick-create-cli
 #
 #  2. Microsoft Azure Key Vault PyPI package -
```
Comment (samvaity) [inline sdk/keyvault/azure-keyvault-keys/samples/hello_world_async.py]: nit: `Azure Key Vault Keys PyPI package`

File: sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py (hunk 1 above)
Comment (chlowell) [reply sdk/keyvault/azure-keyvault-keys/tests/test_examples_keys_async.py]: Removed that yesterday with #6063 😎

**Pull request merged**